test/NuGet.Core.FuncTests/NuGet.XPlat.FuncTest/Package/Download/PackageDownloadRunnerTests.cs

```diff
@@ -0,0 +1,283 @@
+        {
+            new List<(string, string)>
+            {
+                ("Contoso.Preview", "1.3.0"),
+                ("Contoso.Preview", "2.0.0-beta.2")
+            },
+            "Contoso.Preview",        // downloadId argument
+            null,                   // downloadVersion argument
+            true,                   // enablePrerelease
+            ("Contoso.Preview", "2.0.0-beta.2")  // expected
+        };
+
+        // chose stable with IncludePrerelease == false
+        yield return new object[]
+        {
+            new List<(string, string)>
+            {
+                ("Contoso.Preview", "1.3.2"),
+                ("Contoso.Preview", "1.3.0"),
+                ("Contoso.Preview", "2.0.0-beta.2")
+            },
+            "Contoso.Preview",        // downloadId argument
+            null,                   // downloadVersion argument
+            false,                   // enablePrerelease
+            ("Contoso.Preview", "1.3.2")  // expected
+        };
+    }
+
+
+    [Theory]
+    [MemberData(nameof(PackageTestData))]
+    public async Task RunAsync_ExplicitVersionFromLocalFolderSource_SucceedsAsync(
+        IReadOnlyList<(string, string)> sourcePackages,
+        string downloadId,
+        string downloadVersion,
+        bool enablePrerelease,
+        (string, string) expectedPackage)
+    {
+        // Arrange
+        using var context = new SimpleTestPathContext();
+        var sourceDir = context.PackageSource;
+        var outputDir = context.WorkingDirectory;
+
+        foreach (var (id, version) in sourcePackages)
+        {
+            await SimpleTestPackageUtility.CreateFullPackageAsync(sourceDir, id, version);
+        }
+
+        var logger = new Mock<ILoggerWithColor>(MockBehavior.Loose);
+        var settings = new Mock<ISettings>(MockBehavior.Loose);
+
+        var args = new PackageDownloadArgs()
+        {
+            Packages = [new PackageWithNuGetVersion { Id = downloadId, NuGetVersion = downloadVersion == null ? null : NuGetVersion.Parse(downloadVersion) }],
+            OutputDirectory = outputDir,
+            IncludePrerelease = enablePrerelease
+        };
+
+
+        // Act
+        var result = await PackageDownloadRunner.RunAsync(
+            args,
+            logger.Object,
+            [new(sourceDir)],
+            settings.Object,
+            CancellationToken.None);
+
+        // Assert
+        result.Should().Be(PackageDownloadRunner.ExitCodeSuccess);
+        var installDir = Path.Combine(outputDir, expectedPackage.Item1.ToLowerInvariant(), expectedPackage.Item2);
+        Directory.Exists(installDir).Should().BeTrue();
+        Directory.EnumerateFiles(installDir, "*.nupkg").Any().Should().BeTrue();
+        File.Exists(Path.Combine(installDir, $"{expectedPackage.Item1.ToLowerInvariant()}.{expectedPackage.Item2}.nupkg")).Should().BeTrue();
+    }
+
+    [Fact]
+    public async Task RunAsync_NoVersion_PicksHighestAcrossMultipleSources()
```

Comment:
nit: you don't have to do it, but in theory you could've set up examples with multiple sources with the data above too, with little tweaks.